src/agentscope/model/_zhipu_model.py

```diff
@@ -0,0 +1,394 @@
+        """
+        content_blocks: List[TextBlock | ToolUseBlock] = []
+        metadata = None
+
+        if not response.choices:
+            return ChatResponse(content=[], usage=None, metadata=None)
+
+        choice = response.choices[0]
+        message = choice.message
+
+        # Handle text content
+        if message.content:
+            content_blocks.append(
+                TextBlock(
+                    type="text",
+                    text=message.content,
+                ),
+            )
+            if structured_model:
+                metadata = _json_loads_with_repair(message.content)
+
+        # Handle tool calls
+        if message.tool_calls:
+            for tool_call in message.tool_calls:
+                try:
+                    input_data = tool_call.function.arguments
+                    if isinstance(input_data, str):
+                        input_data = _json_loads_with_repair(input_data)
+
+                    content_blocks.append(
+                        ToolUseBlock(
+                            type="tool_use",
+                            id=tool_call.id,
+                            name=tool_call.function.name,
+                            input=input_data,
+                        ),
+                    )
+                except Exception as e:
+                    logger.warning(f"Error parsing tool call: {e}")
+
+        # Calculate usage
+        usage = None
+        if hasattr(response, "usage") and response.usage:
+            usage = ChatUsage(
+                input_tokens=response.usage.prompt_tokens or 0,
+                output_tokens=response.usage.completion_tokens or 0,
+                time=(datetime.now() - start_datetime).total_seconds(),
+            )
+
+        parsed_response = ChatResponse(
+            content=content_blocks,
+            usage=usage,
+            metadata=metadata,
+        )
+
+        return parsed_response
+
+    def _format_tools_json_schemas(
+        self,
+        schemas: list[dict[str, Any]],
+    ) -> list[dict[str, Any]]:
+        """Format the tools JSON schemas to the Zhipu AI format."""
+        return schemas
+
+    def _format_tool_choice(
+        self,
+        tool_choice: str,
+    ) -> str | dict[str, Any]:
+        """Format tool choice to Zhipu AI format."""
+        if tool_choice in ["auto", "none"]:
+            return tool_choice
+        elif tool_choice == "any":
+            # Zhipu AI doesn't support "any", use "auto" instead
+            logger.warning(
+                "Zhipu AI doesn't support tool_choice='any', using 'auto' instead.",
+            )
+            return "auto"
+        elif tool_choice == "required":
+            # Zhipu AI doesn't support "required", use "auto" instead
+            logger.warning(
+                "Zhipu AI doesn't support tool_choice='required', using 'auto' instead.",
+            )
+            return "auto"
+        else:
+            # Specific tool name
```

Comment:
It seems the Zhipu API doesn't support selecting specific tools to execute, only supporting `auto` mode. So we should also raise a warning here.